commit 7556a27e92d5456eeb95b1ca9791f8adfc680e1f
Author: Dmitry Grinberg <dmitrygr@gmail.com>
Date:   Mon Dec 26 23:51:13 2022 -0600

    avoid pushig/popping of window state, colors when not avail

diff --git a/Src/PkmnType.c b/Src/PkmnType.c
index e35ea3a..feff1a1 100644
--- a/Src/PkmnType.c
+++ b/Src/PkmnType.c
@@ -82,6 +82,9 @@ static UInt16 CalculateEffectivenessForType(UInt8 *pkmnBytes, UInt16 typeNum)
 
 static void DrawEffectiveness(UInt16 selectedPkmnID, UInt8 x, UInt8 y, UInt8 typeNum)
 {
+	UInt32 romVersion;
+	IndexedColorType prevColor = 0;
+	FontID prevFont;
 	Char *str;
 	UInt16 effectiveness;
 	MemHandle pInfHndl;
@@ -96,13 +99,23 @@ static void DrawEffectiveness(UInt16 selectedPkmnID, UInt8 x, UInt8 y, UInt8 typ
 	
 	MemHandleUnlock(pInfHndl);
 
+	if (errNone != FtrGet(sysFtrCreator, sysFtrNumROMVersion, &romVersion))
+		romVersion = 0;
+
+	prevFont = FntSetFont(stdFont);
+	if (romVersion >= sysMakeROMVersion(3,5,0,sysROMStageRelease,0)) {
+		
+		prevColor = WinSetTextColor(0);	//get current color
+		WinSetTextColor(prevColor);
+	}
+	
 	if (effectiveness != 100){
-		WinPushDrawState();
 		FntSetFont(boldFont);
 		
 		rgb = GetRGBForEff(effectiveness);
 		
-		WinSetTextColor(WinRGBToIndex(&rgb));
+		if (romVersion >= sysMakeROMVersion(3,5,0,sysROMStageRelease,0))
+			WinSetTextColor(WinRGBToIndex(&rgb));
 	}
 
     x += 35;
@@ -127,6 +140,10 @@ static void DrawEffectiveness(UInt16 selectedPkmnID, UInt8 x, UInt8 y, UInt8 typ
     
 		MemPtrFree(str);
 	}
+	
+	FntSetFont(prevFont);
+	if (romVersion >= sysMakeROMVersion(3,5,0,sysROMStageRelease,0))
+		WinSetTextColor(prevColor);
 }
 
 static void DrawTypeIcons(UInt16 selectedPkmnID)
